lib/Schema.ts

```diff
@@ -582,7 +582,8 @@ export class Schema {
 		}
 
 		this.attributes().forEach((key) => {
-			if (key.includes(".") && this.getAttributeSettingValue("index", key)) {
+			const attributeSettingValue = this.getAttributeSettingValue("index", key);
+			if (key.includes(".") && attributeSettingValue && !Array.isArray(attributeSettingValue)) {
```

Comment:
So the problem with this is that in the event you have an index with multiple types, it won't throw an error. For example, the following schema should throw an error `Index must be at root object and not nested in object or array.`.

```js
new dynamoose.Schema({
    "id": String,
    "obj": {
        "type": Object,
        "schema": {
            "data": [
                {"type": Array, "schema": [String]},
                {"type": String, "index": true}
            ]
        }
    }
});
```

However, with the current code, it doesn't look like it'll throw an error. I think the solution is something like:

```suggestion
			const attributeSettingValue = this.getAttributeSettingValue("index", key);
			if (key.includes(".") && (Array.isArray(attributeSettingValue) ? attributeSettingValue.some((singleValue) => Boolean(singleValue)) : attributeSettingValue)) {
```

That way if it's an array, we are checking to see if `index` exists on any of those types.

Reply:
For sure. Apologizes for missing this, it was difficult to jump into dynamoose and get my bearings quickly. There's a ton of complexity everywhere throughout the code. Impressed seeing the work as I was looking through.

Reply:
@bobbyrc No problem at all! It's not easy jumping into a new codebase. No need at all to apologize for that. Truly appreciate your help on this!! It only gets easier from here haha.